typescript/infra/scripts/core.ts

```diff
@@ -40,6 +50,31 @@ async function main() {
     console.info('Could not load partial core addresses, file may not exist');
   }
 
+  if (argv.fork) {
+    await useLocalProvider(multiProvider, argv.fork);
```

Comment:
Couldn't you just set useLocalProvide before line 33 and then reuse the contract wrapper creation above?

Reply:
yes but I am not passing the previousContracts into the same method